feature/login/src/main/java/com/epmedu/animeal/login/profile/presentation/FinishProfileScreenUi.kt

```diff
@@ -0,0 +1,186 @@
+                keyboardActions = KeyboardActions(onNext = { focusManager.moveFocus(FocusDirection.Down) })
+            )
+            TextInputField(
+                title = stringResource(id = R.string.profile_email_title),
+                hint = stringResource(id = R.string.profile_email_hint),
+                onValueChange = {
+                    viewModel.onEvent(ProfileEvent.EmailChanged(it))
+                },
+                value = state.email,
+                errorText = state.emailError?.asString(),
+                onClearFocus = {
+                    if (state.email.isNotBlank()) {
+                        viewModel.onEvent(ProfileEvent.EmailValidation)
+                    }
+                },
+                keyboardOptions = KeyboardOptions(
+                    imeAction = ImeAction.Done,
+                    keyboardType = KeyboardType.Email
+                ),
+                keyboardActions = KeyboardActions(onDone = { focusManager.clearFocus() })
+            )
+            PhoneNumberInput(
+                title = stringResource(id = R.string.profile_phone_number),
+                value = "999112233",
+                enabled = false
+            )
+            BirthDateInputUi(viewModel, focusManager)
+            Box(
+                modifier = Modifier
+                    .fillMaxWidth()
+                    .padding(top = 8.dp),
+                contentAlignment = Alignment.BottomEnd,
+            ) {
+                AnimealShortButton(
+                    text = stringResource(id = R.string.done),
+                    onClick = {
+                        viewModel.onEvent(ProfileEvent.Submit)
+                        focusManager.clearFocus()
+                    },
+                    modifier = Modifier.padding(start = 44.dp, bottom = 55.dp)
+                )
+            }
+        }
+    }
+}
+
+@Composable
+private fun BirthDateInputUi(viewModel: FinishProfileViewModel, focusManager: FocusManager) {
+    val state = viewModel.state
+    val shouldOpenDialog = remember { mutableStateOf(false) }
+    BirthDateInput(
+        title = stringResource(id = R.string.profile_birth_date),
+        isEnabled = true,
+        onIconClick = {
+            focusManager.clearFocus()
+            shouldOpenDialog.value = true
+        },
+        onValueChange = {
+            viewModel.onEvent(ProfileEvent.BirthDateChanged(it))
+        },
+        value = state.birthDateString,
+        errorText = state.birthDateError?.asString(),
+        keyboardOptions = KeyboardOptions(
+            imeAction = ImeAction.Done,
+            keyboardType = KeyboardType.Number
+        ),
+        keyboardActions = KeyboardActions(onDone = {
+            ProfileEvent.BirthDateValidation
+            focusManager.clearFocus()
+        })
+    )
+    DatePickerDialog(
+        initialDate = viewModel.state.initialDate,
+        shouldShowDialog = shouldOpenDialog
```

Comment:
you can write `val shouldOpenDialog by remember { mutableStateOf(false) }` 
and you will simplify code a little bit

Reply:
Done